Client/YooChooseNotifier.php

```diff
@@ -129,9 +129,9 @@ public function setIncludedContentTypes($value)
      */
     public function updateContent($contentId, $versionNo = null)
     {
-        $content = $this->contentService->loadContent($contentId, null, $versionNo);
-
         try {
+            $content = $this->contentService->loadContent($contentId, null, $versionNo);
```

Comment:
I feel like this is nothing more than covering the real issue. The issue is that you are doing something counter intuitive - loading a content in version which has been removed recently.  You could've just removed `deleteVersion` slot completely as the result would be the same. With try-catch you are also hiding potential other issues as well. 

I don't know YC API at all and can't tell how they recommender system should be notified about removed version. Does it even care about removed draft version?

Reply:
This PR is reverting previous behaviour, I've made a mistake in cleanup, the loadContent method should be, and it was inside the try/catch before [link](https://github.com/ezsystems/EzSystemsRecommendationBundle/commit/9a86ca8e99d216130461ec926371e815d1d31fb7#diff-99985165fd97be8057bac2eb7e127481L134)
(loadContent is called by getContentTypeIdentifier method)

Current dataflow needs to be reviewed and covered by extended tests, but as another PR

Reply:
Sounds reasonable to me as a quick fix but indeed it's a good candidate for solid rework.